### PR TITLE
[SW-949] Add Mission Client in spot_wrapper

### PR DESCRIPTION
**Pull request by tcappellari-bdai**
Added mission client into `spot_wrapper`
Made new `spot_mission_wrapper.py` to load, play, pause, restart, stop, and get the state of a mission.
Added `MissionClient` to the sdk to start the service (not a part of the default services in `bosdyn/client/sdk.py`)

### CODE REVIEW

**Comment by tcappellari-bdai** [comment]
* **#105** <a href="https://app.graphite.dev/github/pr/bdaiinstitute/spot_wrapper/105?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tcappellari-bdai and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: spot_wrapper/spot_mission_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+          leases: Leases the mission service will need to use. Unlike other clients, these MUST
+              be specified.
+        Raises:
+            RpcError: Problem communicating with the robot.
+            NoMissionError: No mission Loaded.
+        """
+        try:
+            resp = self._mission_client.play_mission_async(pause_time_secs, leases, settings)
+        except RpcError:
+            resp = (False, "Could not communicate with the robot")
```
Comment (amessing-bdai) [inline spot_wrapper/spot_mission_wrapper.py:121]: Any reason to store this versus just return it?

File: spot_wrapper/spot_mission_wrapper.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        except NoMissionError:
+            resp = (False, "No mission loaded")
+        return resp
+
+    def _get_mission_state(self, upper_tick_bound: int = None, lower_tick_bound: int = None, past_ticks: int = None):
```
Comment (amessing-bdai) [inline spot_wrapper/spot_mission_wrapper.py:126]: parameters should be `Optional` if they can accept `None`

File: spot_wrapper/spot_mission_wrapper.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        """
+        try:
+            return self._mission_client.get_info()
+        except RpcError:
+            return False, "Could not communicate with the robot"
+
+    def _play_mission(
+        self,
+        pause_time_secs: int,
+        leases=[],
```
Comment (amessing-bdai) [inline spot_wrapper/spot_mission_wrapper.py:104]: You shouldn't use a mutable type as a default parameter. You should have the default be `None`, then internal to the function you can check if `None` and set as `[]`

File: spot_wrapper/spot_mission_wrapper.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+        self._robot_command_client = robot_command_client
+        self._lease_client = lease_client
+        self._robot_state = robot_state
+        self._spot_check_resp = None
+        self._lease = None
+        self._lease_wallet: LeaseWallet = self._lease_client.lease_wallet
+
+    def _get_lease(self) -> Lease:
+        self._lease = self._lease_wallet.get_lease()
+        return self._lease
```
Comment (amessing-bdai) [inline spot_wrapper/spot_mission_wrapper.py:44]: Do you use this anywhere?

File: spot_wrapper/spot_mission_wrapper.py [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+
+        Raises:
+            RpcError: Problem communicating with the robot.
+        """
+        try:
+            return self._mission_client.get_info()
+        except RpcError:
+            return False, "Could not communicate with the robot"
+
+    def _play_mission(
```
Comment (amessing-bdai) [inline spot_wrapper/spot_mission_wrapper.py:101]: Why are all of these functions private? I would assume you play to use them externally in spot_ros2?

File: spot_wrapper/spot_mission_wrapper.py (hunk 1 above)
Comment (tcappellari-bdai) [reply spot_wrapper/spot_mission_wrapper.py:121]: not beyond a stylistic choice to have the return at the end. I can change it to just return directly instead

**Review by khughes-bdai — APPROVED** [review]
LGTM

**Pull request merged**